tools/SetupFlow/DevHome.SetupFlow/ViewModels/QuickstartPlaygroundViewModel.cs

```diff
@@ -430,73 +446,102 @@ public void UpdateProgress(QuickStartProjectProgress progress)
-            IsFileViewVisible = false;
-            IsErrorViewVisible = false;
+                TelemetryFactory.Get<ITelemetry>().Log("QuickstartPlaygroundGenerateButtonClicked", LogLevel.Critical, new GenerateButtonClicked(userPrompt), ActivityId);
 
-            // Ensure that the launch buttons are in their default state. This is important for scenarios where
-            // the extension has more than one project host and the user is doing multiple generate attempts in sequence.
-            // It makes sure that the code in the code-behind for populating the dropdown button gets re-run to pick up
-            // the new project host objects.
-            IsLaunchButtonVisible = true;
-            IsLaunchDropDownVisible = false;
+                // Ensure file view isn't visible (in the case where the user has previously run a Generate command)
+                IsFileViewVisible = false;
+                IsErrorViewVisible = false;
 
-            // Without this, when the user generates two projects in sequence, the text box
-            // will contain any text from last-opened file in the previous project (which is
-            // confusing as this project may not have anything to do with the current one). This
-            // ensures that the file view is back to a known, clean state.
-            GeneratedFileContent = string.Empty;
+                // Ensure that the launch buttons are in their default state. This is important for scenarios where
+                // the extension has more than one project host and the user is doing multiple generate attempts in sequence.
+                // It makes sure that the code in the code-behind for populating the dropdown button gets re-run to pick up
+                // the new project host objects.
+                IsLaunchButtonVisible = true;
+                IsLaunchDropDownVisible = false;
 
-            // Temporarily turn off the provider combobox and ensure user cannot edit the prompt for the moment
-            EnableQuickstartProjectCombobox = false;
-            IsPromptTextBoxReadOnly = true;
-            EnableProjectButtons = false;
+                // Without this, when the user generates two projects in sequence, the text box
+                // will contain any text from last-opened file in the previous project (which is
+                // confusing as this project may not have anything to do with the current one). This
+                // ensures that the file view is back to a known, clean state.
+                GeneratedFileContent = string.Empty;
 
-            IProgress<QuickStartProjectProgress> progress = new Progress<QuickStartProjectProgress>(UpdateProgress);
+                // Temporarily turn off the provider combobox and ensure user cannot edit the prompt for the moment
+                EnableQuickstartProjectCombobox = false;
+                IsPromptTextBoxReadOnly = true;

```

Comment:
Please stay consistent with what `true` means.

In this case `false` will disable two controls and `true` disables another.  Consider renaming `IsPromptTextBoxReadOnly` in  a way where `false` makes it readonly.

Considering all these controls need to be disabled.  Can these three controls be controlled with 1 bool instead of three?